accepted/00000-jsonb-pillar-db-storage.md

```diff
@@ -0,0 +1,153 @@
+- Feature Name: Database backed salt pillar storage
+- Start Date: 2021-01-13
+- RFC PR: TBD
+
+# Summary
+[summary]: #summary
+
+Use database as a storage for Uyuni salt pillars instead of plain files.
+
+# Motivation
+[motivation]: #motivation
+
+Pillars are used for passing reliable and trustworthy data to salt clients. Salt integration into Uyuni generates pillar files which are then loaded to the salt ecosystem by Uyuni external pillar.
+Advantage of this solution is rather easy implementation and quick visibility of the data when inspecting in filesystem, however there are disadvantages to consider. Data consistency [issue#5678](https://github.com/SUSE/spacewalk/issues/5678) and [issue#10679](https://github.com/SUSE/spacewalk/issues/10679), [filesystem performance](https://github.com/SUSE/spacewalk/pull/10391#discussion_r375147529), backup procedure must consider these files as well.
```

Comment:
The file approach was choosen as the original plan was, to use salt-syndic on the Proxies. One requirement of syndic is, that all states and pillars are availabel on the syndic. Transfer files or use a network storage would be required for states anyway. So pillar as files was just the same problem and easier then opening database ports to the world and dealing with a second access way.

As the syndic story seems not to become reality, putting pillar into the DB can make things easier in some areas.